frontend/src/containers/OnBoarding.js

```diff
@@ -55,12 +54,17 @@ export class OnBoarding extends Component {
       githubUsername,
       fetchReposFromGitHub,
       appendGithubForm,
-      params
+      params,
+      notify
     } = this.props;
 
-    if (githubUsername && this.state.step === 2) {
+    if (githubUsername && this.state.step === 1) {
       appendGithubForm({username: githubUsername, token: params.token});
-      fetchReposFromGitHub(githubUsername);
+      fetchReposFromGitHub(githubUsername)
+      .catch((error) => {
+        this.setState({step: 0});
```

Comment:
I don't understand how it will loop infinitely, there is an `if` check that tests the step is one on line 61
I've manually tested it without any infinite loops